internal/controller/pipelinerollout_controller.go

```diff
@@ -197,28 +197,31 @@ func (r *PipelineRolloutReconciler) reconcile(
 		return false, fmt.Errorf("error getting Pipeline: %v", err)
 	}
 
+	// propagate the pipeline's status into PipelineRollout's status
+	processPipelineStatus(ctx, existingPipelineDef, pipelineRollout)
```

Comment:
I think it should actually be okay to do this early on in the function. Since we watch the underlying child resource, if a child's status changes based on any changes we make to the child's spec, that will make another call to `Reconcile()` and we will process it then.